recipes/cpython/all/test_package/conanfile.py

```diff
@@ -18,175 +25,200 @@ def __init__(self, debug, pymalloc, unicode):
 
     def test(self):
-        if not tools.cross_building(self, skip_x64_x86=True):
-            self.run("{} -c \"print('hello world')\"".format(self.deps_user_info["cpython"].python), run_environment=True)
+        if can_run(self):
+            self.run(f"{self._python} --version", env="conanrun")
+
+            self.run(f"{self._python} -c \"print('hello world')\"", env="conanrun")
 
             buffer = StringIO()
-            self.run("{} -c \"import sys; print('.'.join(str(s) for s in sys.version_info[:3]))\"".format(self.deps_user_info["cpython"].python), run_environment=True, output=buffer)
+            self.run(f"{self._python} -c \"import sys; print('.'.join(str(s) for s in sys.version_info[:3]))\"", buffer, env="conanrun")
             self.output.info(buffer.getvalue())
             version_detected = buffer.getvalue().splitlines()[-1].strip()
             if self._py_version != version_detected:
-                raise ConanException("python reported wrong version. Expected {exp}. Got {res}.".format(exp=self._py_version, res=version_detected))
+                raise ConanException(
+                    f"python reported wrong version. Expected {self._py_version}. Got {version_detected}."
+                )
 
             if self._supports_modules:
                 self._test_module("gdbm", self._cpython_option("with_gdbm"))
                 self._test_module("bz2", self._cpython_option("with_bz2"))
-                self._test_module("bsddb", self._cpython_option("with_bsddb"))
+                if self._py_version.major < 3:
+                    self._test_module("bsddb", self._cpython_option("with_bsddb"))
                 self._test_module("lzma", self._cpython_option("with_lzma"))
                 self._test_module("tkinter", self._cpython_option("with_tkinter"))
-                with tools.environment_append({"TERM": "ansi"}):
-                    self._test_module("curses", self._cpython_option("with_curses"))
+                os.environ["TERM"] = "ansi"
+                self._test_module("curses", self._cpython_option("with_curses"))
 
                 self._test_module("expat", True)
-                self._test_module("sqlite3", True)
+                self._test_module("sqlite3", self._cpython_option("with_sqlite3"))
                 self._test_module("decimal", True)
                 self._test_module("ctypes", True)
-
-            if tools.is_apple_os(self.settings.os) and not self.options["cpython"].shared:
-                self.output.info("Not testing the module, because these seem not to work on apple when cpython is built as a static library")
+                skip_ssl_test = is_msvc(self) and self._py_version < "3.8" and self._cpython_option("shared")
+                if not skip_ssl_test:
+                    # Unsure cause of failure in this oddly specific combo, but these versions are EOL so not concerned with fixing.
+                    self._test_module("ssl", True)
```

Comment:
```suggestion
                    env = Environment()
                    if self.settings.os != "Windows":
                        env.define_path("OPENSSL_CONF", os.path.join(os.sep, "dev", "null"))
                    with env.vars(self).apply():
                        self._test_module("ssl", True)
```

Reply:
Will also apply this shortly, just getting to the other issue.